vim-patch:8.1.{2355,2363}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.1.2355: test with "man" fails on FreeBSD

Problem:    Test with "man" fails on FreeBSD.
Solution:   Use "-P" instead of "--pager".
https://github.com/vim/vim/commit/c7d2a57b3a076f6ecb16f93c0b09280c4b3b4175

Cherry-picked "has('bsd')" from vim-patch:8.1.0846.
Cherry-picked test_normal.vim fix from vim-patch:8.1.2358

blueyed reviewed (approved):
Note that it does not appear to be a problem for us though?
(not included with the failures there on sourcehut: https://github.com/neovim/neovim/pull/11477#discussion_r352095005)

janlazo commented on src/nvim/eval.c:
```diff
@@ -11503,6 +11503,9 @@ static void f_glob2regpat(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 static void f_has(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
   static const char *const has_list[] = {
+#if defined(BSD) && !defined(__APPLE__)
```
Vim uses `MACOS_X`. Should `has('bsd')` work on OSX?

justinmk replied:
> Should `has('bsd')` work on OSX?

I'd say no, otherwise it makes `has('bsd')` useless...

justinmk commented:
@janlazo could you add a line to `:help feature-list` ? Looks like "mac" is missing there too. (NB: that list is intentionally much smaller than Vim's)

janlazo committed:
doc: mention OS pseudo-features in :h feature-list

janlazo committed:
vim-patch:8.1.2363: ml_get error when accessing Visual area in 'statusline'

Problem:    ml_get error when accessing Visual area in 'statusline'.
Solution:   Disable Visual mode when using another window. (closes vim/vim#5278)
https://github.com/vim/vim/commit/dee50a518007b3a59f54b8ad018b6a83993593e7

justinmk commented on runtime/doc/eval.txt:
```diff
@@ -4969,10 +4969,15 @@ has({feature})	Returns 1 if {feature} is supported, 0 otherwise.  The
+			macunix		Synonym for mac.
 			nvim		This is Nvim.
+			osx		Synonym for mac.
+			osxdarwin	Synonym for mac.
```
I intentionally elided variants like "osx", "win64" since they're just noise.

justinmk commented on runtime/doc/eval.txt:
```diff
@@ -4969,10 +4969,15 @@ has({feature})	Returns 1 if {feature} is supported, 0 otherwise.  The
 <							*feature-list*
 		    List of supported pseudo-feature names:
 		        acl		|ACL| support
+			bsd		BSD system.
```
```suggestion
			bsd		BSD system (not macOS, use "mac" for that).
```
